src/core/SIP/SIPMessageBuffer.cs

```diff
@@ -55,50 +69,81 @@ public class SIPMessageBuffer
-                if (buffer == null || buffer.Length < m_minFirstLineLength)
-                {
-                    // Ignore.
-                    return null;
-                }
-                else if (buffer.Length > SIPConstants.SIP_MAXIMUM_RECEIVE_LENGTH)
-                {
-                    throw new ApplicationException("SIP message received that exceeded the maximum allowed message length, ignoring.");
-                }
-                else if (!BufferUtils.HasString(buffer, 0, buffer.Length, SIP_MESSAGE_IDENTIFIER, m_CRLF))
-                {
-                    // Message does not contain "SIP" anywhere on the first line, ignore.
-                    return null;
-                }
-                else
-                {
-                    message = Encoding.UTF8.GetString(buffer, 0, buffer.Length);
-                    SIPMessageBuffer sipMessageBuffer = ParseSIPMessage(message, localSIPEndPoint, remoteSIPEndPoint);
-
-                    if (sipMessageBuffer != null)
-                    {
-                        sipMessageBuffer.RawBuffer = buffer;
-                        return sipMessageBuffer;
-                    }
-                    else
-                    {
-                        return null;
-                    }
-                }
-            }
-            catch (Exception excp)
-            {
-                message = message.Replace("\n", "LF");
-                message = message.Replace("\r", "CR");
-                logger.LogError("Exception ParseSIPMessage. " + excp.Message + "\nSIP Message=" + message + ".");
-                return null;
+        {
+
+            if (buffer == null || buffer.Length < m_minFirstLineLength)
+            {
+                // Ignore.
+                return null;
+            }
+            else if (buffer.Length > SIPConstants.SIP_MAXIMUM_RECEIVE_LENGTH)
+            {
+                throw new ApplicationException("SIP message received that exceeded the maximum allowed message length, ignoring.");
+            }
+            else if (!BufferUtils.HasString(buffer, 0, buffer.Length, SIP_MESSAGE_IDENTIFIER, m_CRLF))
+            {
+                // Message does not contain "SIP" anywhere on the first line, ignore.
+                return null;
+            }
+            else
+            {
+                var sipMessage = new SIPMessageBuffer();
+
+                sipMessage.RawBuffer = buffer;
+                sipMessage.LocalSIPEndPoint = localSIPEndPoint;
+                sipMessage.RemoteSIPEndPoint = remoteSIPEndPoint;
+
+                // For connection oriented transports the same connection should be used for responses and subsequent requests.
+                if (sipMessage.LocalSIPEndPoint != null && remoteSIPEndPoint.ConnectionID != null)
+                {
+                    sipMessage.LocalSIPEndPoint.ConnectionID = remoteSIPEndPoint.ConnectionID;
+                }
+
+                string message = Encoding.UTF8.GetString(buffer);

```

Comment:
For performance. The buffer might be quite large. Search the byte array for endHeaderPosn and utf8 decode to the end of headersection only.

Reply:
Yes that would be a good improvement but see point above about string processing not being a significant bottleneck. Another good target for a subsequent PR.